### PR TITLE
[Fix] use function name inference and computed properties to distinguish `Symbol()` from `Symbol(‘’)`, when available

### LINKED ISSUE #1: Prefer function .name to overwriting Symbol
```js
let empty = Symbol('');
let undef = Symbol();
console.log(
  {[empty](){}}[empty].name === '[]', // true
  {[undef](){}}[undef].name === '', // true
);
```

Except this only works on browsers where function name inference works correctly for symbol-named functions, which in my tests includes Chrome and FF but not Safari.

### PR DESCRIPTION
**Pull request by ljharb**
Fixes #1.

cc @bakkot @michaelficarra

### CODE REVIEW

**Review by bakkot — APPROVED** [review]
Would strongly prefer that `shimGlobal` do nothing when `getInferredName` is available, personally. Don't touch the globals if you don't absolutely need to.

Otherwise LGTM.

**Comment by ljharb** [comment]
Good point; I’ll add tests that cover when getInferredName is available that ensure that Symbol is not shimmed.

**Comment by ljharb** [comment]
Updated!

**Review by bakkot — APPROVED** [review]

File: helpers/getInferredName.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+var getInferredName;
+try {
+	// eslint-disable-next-line no-new-func
+	getInferredName = Function('s', 'return { [s]() {} }[s].name;');
```
Comment (michaelficarra) [inline helpers/getInferredName.js:6]: Why does this use the Function constructor on strings instead of just a function expression?

Comment (ljharb) [reply helpers/getInferredName.js:6]: because that would be a syntax error if the engine didn't support the syntax; that's what the try/catch is for.

**Review by michaelficarra — APPROVED** [review]

**Pull request merged**